src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectStrategy.java

```diff
@@ -53,5 +59,35 @@
      * @param item the item in queue, which will be a build.
      * @return
      */
-    public abstract Authentication authenticate(AbstractProject<?, ?> project, Queue.Item item);
+    public Authentication authenticate(Job<?, ?> project, Queue.Item item) {
+        if(!Util.isOverridden(
+                AuthorizeProjectStrategy.class,
+                getClass(),
+                "authenticate",
+                AbstractProject.class,
+                Queue.Item.class
+        )) {
+            throw new AbstractMethodError();
+        }
+        
+        if (!(project instanceof AbstractProject)) {
+            Descriptor<?> d = Jenkins.getInstance().getDescriptor(getClass());
+            LOGGER.log(
+                    Level.WARNING,
+                    "This authorization strategy ({0}) is designed for authorize-project < 1.1.0 and not applicable for non-AbstractProjects (like WorkflowJob). ignored.",
+                    (d != null)?d.getDisplayName():getClass().getName()
+            );

```

Comment:
Created https://issues.jenkins-ci.org/browse/JENKINS-28258 (just for the future use)